public/javascripts/SVValidate/src/status/StatusField.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Updates items that appear on the right side of the validation interface (i.e., label counts)
+ * @param missionMetadata   Metadata for the initial mission
+ * @returns {StatusField}
+ * @constructor
+ */
+function StatusField(missionMetadata) {
+    var self = this;
+
+    var labelNames = {
+        CurbRamp : "Curb Ramp",
+        NoCurbRamp : "Missing Curb Ramp",
+        Obstacle : "Obstacle in Path",
+        SurfaceProblem : "Surface Problem",
+        NoSidewalk : "Missing Sidewalk",
+        Other : "Other"
+    };
+
+    /**
+     * Resets the status field whenever a new mission is introduced.
+     * @param currentMission    Mission object for the current mission.
+     */
+    function reset(currentMission) {
+        var progress = currentMission.getProperty('labelsProgress');
+        var total = currentMission.getProperty('labelsValidated');
+        var completionRate = progress / total;
+        updateLabelCounts(progress);
+        updateMissionDescription(total);
+        setProgressText(completionRate);
+        setProgressBar(completionRate);
+    }
+
+    /**
+     * Updates the number of labels the user has validated.
+     * @param count {int} Number of labels the user has validated.
+     */
+    function updateLabelCounts(count) {
+        svv.ui.status.labelCount.html(count);
+    }
+
+    /**
+     * Updates the label name that is displayed in the status field.
+     * Updates the label name that is displayed in the title bar.
```

Comment:
Does this do both of these things or only one of them..? Can't tell if this is a copy/paste mistake or was done on purpose.

Reply:
It does both, but I agree it's confusing. Updated the comment so hopefully it's clearer.